src/rdb.c

```diff
@@ -124,7 +124,7 @@ int rdbLoadType(rio *rdb) {
  * to check for loading errors the caller should call rioGetReadError() after
  * calling this function. */
 time_t rdbLoadTime(rio *rdb) {
-    int32_t t32;
+    int64_t t32;
     if (rioRead(rdb,&t32,4) == 0) return -1;
     return (time_t)t32;
```

Comment:
Why read 4 bytes into an 8 bytes type?

Reply:
It is written as 4 bytes in the RDB file so it must be read back the same way. To solve it, we would need to write it as 8 bytes in the RDB file, but that is not a backwards compatible change.

Reply:
I found that in 'rdbSaveKeyValuePair' function, expiretime is using 8 bytes, but in 'rdbLoadTime' function, expiretime is using 4 bytes

Reply:
And in the 'rdbGenericLoadStringObject' function, reading supports up to 4 bytes.Redis uses 8 bytes when generating rdb, but uses 4 bytes when restoring data by restarting redis to load rdb.

Reply:
Read the comment before the function:

```C
/* This is only used to load old databases stored with the RDB_OPCODE_EXPIRETIME
 * opcode. New versions of Redis store using the RDB_OPCODE_EXPIRETIME_MS
 * opcode. On error -1 is returned, however this could be a valid time, so
 * to check for loading errors the caller should call rioGetReadError() after
 * calling this function. */
time_t rdbLoadTime(rio *rdb) {
```

Saving expiretime as 8 bytes with millisecond precision was introduced in 2011 in commit 7dcc10b65e.

Reply:
obviously, if we write 8 bytes, and we read 4, then the rest of the file won't be parsed correctly.
so @Kalone-Lin maybe you're missing something, or maybe you found something but we still don't understand what it is.
please explain better, or if you realize you're wrong, close the issue.

p.s. reading 4 bytes into a 64bit variable (what your PR currently does), will have endianess issues.